drivers/bluetooth/hci/h4.c

```diff
@@ -423,20 +419,35 @@ static int h4_send(struct net_buf *buf)
 	return 0;
 }
 
+/** Setup the HCI transport, which usually means to reset the Bluetooth IC
+  *
+  * @param dev The device structure for the bus connecting to the IC
+  *
+  * @return number of additional bytes to discard, negative error
+  * 	    value on failure
+  */
+int __weak bt_hci_transport_setup(struct device *dev)
+{
+	/* Return a value > 0 by default so that h4_open knows to discard
+	 * any bytes, since this function does nothing. */
+	return 32;
```

Comment:
This return value meaning seems very specific to UART. I'm not sure it's possible to justify that this is an obvious choice regardless of the transport used (or that no other info needs to be returned by other transports). I wouldn't be that much bothered by this, but the way you've named the callback is completely generic and should apply to any transport. Would it be possible to make the return value void and then handle the discarding internally in the setup function?

Reply:
To put this another way: it should be possible to add calls to `bt_hci_transport_setup()` to all existing HCI drivers' `open()` callbacks in a meaningful way. In fact, it'd be nice if this PR did that for completeness.

Reply:
Yep, I can do that!